README.md

```diff
@@ -1,2 +1,95 @@
+    - Personal blogs that link to affiliated VR sites or your friends blogs.
+    - Educational websites for a class, that link to the school's portal or other resources.
+    - User generated content linking to one another. Like Rec Room, VR Chat or Super Craft.
+    - Sponsored links. You have a VR world about photography and filmmaking that links to Photorama, B&H, GoPro's VR portals.
+    - Content portals in general (directories of games, immersive videos, collections).
+    - Personal storefronts that link to products hosted by eBay VR, Amazon VR. One could also easily incorporate product 3D models hosted externally in services the like of Sketchfab or Google Blocks.
+
+2. In an AR browser, virtual objects might be fetched from different URLs and rendered in physical context. Content is expected to always present at load time.
+
+3. Open in immersive mode. 2D windowed browsers might offer a way to open a link directly immersive mode, similar to "open in new tab" or "open in new page" actions offered today.
+
+4. Intents, deep linking, PWA, kiosk mode. Native aplications might request the browser to open content directly in immersive mode.
+
+5. Performance implications of single page applications. Having different web pages load is sort of beneficial for not having to deal with offloading / onloading new models and environments or making your own interstitial. You can avoid frame drops by letting the browser handle the transition, and VR web pages load quick enough.
+
+## API Proposal
+
+UAs fire a `sessiongranted` event that allows content to enter immersive mode without user gesture activation:
+
+```
+navigator.xr.addEventListener('sessiongranted', function (evt) {
+   // One could check for the type of session granted.
+   // Events notifies of session creation after navigation, UA action, or requestSession.
+   // The session object is provided as part of this event.
+   if (evt.session.mode === 'immersive-vr') {
+      // set up app state for immersive vr, if that's what the app wants
+   } else {
+      // notify user that this app only works in immersive vr mode, if desired
+   }
+}
+```
+
+This is similar to the [WebVR displayactivate event](https://immersive-web.github.io/webvr/spec/1.1/#dom-window-onvrdisplayactivate) that popular libraries and frameworks incorporate alongide user gesture activated flows. The UA always remains in control on when immersive sessions are granted and can also revoke them.
+
+## Security considerations
+
+In immersive mode the Browser UI may not be rendered and content controls all display real estate. The UA must steward users giving them information and control over displayed content at all times.
+
+Spoofing is a potential risk during immersive navitagion. A malicious site could masquerade legitimate content, the browser UI or even the underling operating system interface without user noticing. Possible mitigaton mechanisms are:
```

Comment:
Might be good to provide more detailed examples of how to handle things.  Any example we give here should either say "unclear right now how to do this" or "here is an explicit way to do that is safe."  We only need _one_ in the later category as an option, but need _at least one_ for this to be viable.  For example, a modal dialog (assuming the platform has a way of presenting modal dialogs that the user would recognize)  as part of the transition would be fine -- it's a bit jaring, but far less so than popping all the way out to 2D.

Reply:
Are the mechanisms below not explicit enough?

Reply:
no, they don't spell out how/why they would _definitely_ be safe.  They are fine as starting points but are vague.